mantle/kola/tests/misc/boot-mirror-luks.go

```diff
@@ -0,0 +1,273 @@
+	*/
+	bootmirrorluks = conf.Ignition(`{
+		"ignition": {
+		  "version": "3.2.0"
+		},
+		"storage": {
+		  "disks": [
+			{
+			  "device": "/dev/vda",
+			  "partitions": [
+				{
+				  "label": "bios-1",
+				  "sizeMiB": 1,
+				  "typeGuid": "21686148-6449-6E6F-744E-656564454649"
+				},
+				{
+				  "label": "esp-1",
+				  "sizeMiB": 127,
+				  "typeGuid": "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+				},
+				{
+				  "label": "boot-1",
+				  "sizeMiB": 384
+				},
+				{
+				  "label": "root-1"
+				}
+			  ],
+			  "wipeTable": true
+			},
+			{
+			  "device": "/dev/vdb",
+			  "partitions": [
+				{
+				  "label": "bios-2",
+				  "sizeMiB": 1,
+				  "typeGuid": "21686148-6449-6E6F-744E-656564454649"
+				},
+				{
+				  "label": "esp-2",
+				  "sizeMiB": 127,
+				  "typeGuid": "C12A7328-F81F-11D2-BA4B-00A0C93EC93B"
+				},
+				{
+				  "label": "boot-2",
+				  "sizeMiB": 384
+				},
+				{
+				  "label": "root-2"
+				}
+			  ],
+			  "wipeTable": true
+			}
+		  ],
+		  "filesystems": [
+			{
+			  "device": "/dev/disk/by-partlabel/esp-1",
+			  "format": "vfat",
+			  "label": "esp1",
+			  "wipeFilesystem": true
+			},
+			{
+				"device": "/dev/disk/by-partlabel/esp-2",
+				"format": "vfat",
+				"label": "esp2",
+				"wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/md/md-boot",
+			  "format": "ext4",
+			  "label": "boot",
+			  "wipeFilesystem": true
+			},
+			{
+			  "device": "/dev/mapper/root",
+			  "format": "xfs",
+			  "label": "root",
+			  "wipeFilesystem": true
+			}
+		  ],
+		  "luks": [
+			{
+			  "clevis": {
+				"tpm2": true
+			  },
+			  "device": "/dev/md/md-root",
+			  "label": "luks-root",
+			  "name": "root",
+			  "wipeVolume": true
+			}
+		  ],
+		  "raid": [
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/boot-1",
+				"/dev/disk/by-partlabel/boot-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-boot",
+			  "options": [
+				"--metadata=1.0"
+			  ]
+			},
+			{
+			  "devices": [
+				"/dev/disk/by-partlabel/root-1",
+				"/dev/disk/by-partlabel/root-2"
+			  ],
+			  "level": "raid1",
+			  "name": "md-root"
+			}
+		  ]
+		}
+	  }`)
+)
+
+func mustMatch(c cluster.TestCluster, r string, output []byte) {
+	m, err := regexp.Match(r, output)
+	if err != nil {
+		c.Fatalf("Failed to match regexp %s: %v", r, err)
+	}
+	if !m {
+		c.Fatalf("Regexp %s did not match text: %s", r, output)
+	}
+}
+
+func mustNotMatch(c cluster.TestCluster, r string, output []byte) {
+	m, err := regexp.Match(r, output)
+	if err != nil {
+		c.Fatalf("Failed to match regexp %s: %v", r, err)
+	}
+	if m {
+		c.Fatalf("Regexp %s matched text: %s", r, output)
+	}
+}
+
+func init() {
+	register.RegisterTest(&register.Test{
+		Run:                  runBootMirrorLUKSTest,
+		ClusterSize:          0,
+		Name:                 `coreos.boot-mirror.luks`,
+		Platforms:            []string{"qemu-unpriv"},
+		ExcludeArchitectures: []string{"s390x"}, // no TPM support for s390x in qemu
```

Comment:
this would fail for other arches like aarch64 and ppc64le too unless the fcct specifies the `layout`. might want to exclude aarch64 and ppc64le or add the layout to the fcct based on architecture.

Reply:
Good catch! I think eventually we should just vendor in fcct so tests can specify FCCs directly. For now, we can just pipe it through `/usr/bin/fcct` which we already support for external tests.